File: elixir/lib/features.ex
# BluefinTecsMerchantServices SDK feature factory

defmodule BluefinTecsMerchantServices.Features do
  def make_feature(name) do
    case name do
      "audit" -> BluefinTecsMerchantServices.Feature.Audit.new()
      "clienttrack" -> BluefinTecsMerchantServices.Feature.Clienttrack.new()
      "idempotency" -> BluefinTecsMerchantServices.Feature.Idempotency.new()
      "log" -> BluefinTecsMerchantServices.Feature.Log.new()
      "metrics" -> BluefinTecsMerchantServices.Feature.Metrics.new()
      "paging" -> BluefinTecsMerchantServices.Feature.Paging.new()
      "ratelimit" -> BluefinTecsMerchantServices.Feature.Ratelimit.new()
      "retry" -> BluefinTecsMerchantServices.Feature.Retry.new()
      "telemetry" -> BluefinTecsMerchantServices.Feature.Telemetry.new()
      "test" -> BluefinTecsMerchantServices.Feature.Test.new()
      "timeout" -> BluefinTecsMerchantServices.Feature.Timeout.new()
      _ -> BluefinTecsMerchantServices.Feature.new()
    end
  end
end

File: py/features.py
# BluefinTecsMerchantServices SDK feature factory

from feature.base_feature import BluefinTecsMerchantServicesBaseFeature
from feature.test_feature import BluefinTecsMerchantServicesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BluefinTecsMerchantServicesBaseFeature(),
        "test": lambda: BluefinTecsMerchantServicesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
